CIP-0PRS/README.lagda.md

```diff
@@ -0,0 +1,1568 @@
+| Chain-quality time      | $T_\text{CQ}$   | slots   | Ensure the presence of at least one honest block on the chain.                            | $T_\text{CQ} = \mathcal{O} (k/f)$                        | A least one honest block must be produced.                                                   |
+| Common-prefix time      | $T_\text{CP}$   | slots   | Achieve settlement.                                                                       | $T_\text{CP} = \mathcal{O} (k/f)$                        | The Ouroboros Praos security parameter defines the time for having a common prefix.          |
+| Security parameter      | $k$             | blocks  | The Ouroboros Praos security parameter.                                                   | n/a                                                      | Value for the Cardano mainnet.                                                               |
+
+
+### Specification of votes and certificates
+
+The stake-proportional voting in Peras mimics the _sortition_ algorithm used in Praos: specifically it is based on the use of a *verifiable random function* (VRF) by each stake-pool operator guaranteeing the following properties:
+
+- The probability for each voter to cast their vote in a given round is correlated to their share of total stake.
+- It should be computationally impossible to predict a given SPO's schedule without access to their secret key VRF key.
+- Verification of a voter's right to vote in a round should be efficiently computable.
+- A vote should be unique and non-malleable, which is a requirement for the use of efficient certificates aggregation.
+
+Additionally one would like the following property to be provided by the voting scheme:
+
+- Voting should require minimal additional configuration (e.g., key management) for SPOs.
+- Voting and certificate construction should be fast in order to ensure it does not interfere with other operations happening in the node.
+
+The precise scheme and format for votes and certificates is immaterial to the protocol itself and is deferred to another proposed CIP [*Votes & Certificates on Cardano*](https://github.com/cardano-foundation/CIPs/pull/870).
+
+### CDDL schema for the ledger
+
+Peras requires a single addition, `peras_cert`, the [block](#blocks) data on the ledger.
+
+```diff
+ block =
+   [ header
+   , transaction_bodies         : [* transaction_body]
+   , transaction_witness_sets   : [* transaction_witness_set]
+   , auxiliary_data_set         : {* transaction_index => auxiliary_data }
+   , invalid_transactions       : [* transaction_index ]
++  , ? peras_cert               : votes_certificate
+   ]
+```
+
+[Votes](#votes) are serialized in the following CDDL.
+
+```cddl
+vote =
```

Comment:
tiny nit pick, but vote is already defined via [Conway CDDL](https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/conway/impl/cddl-files/conway.cddl#L506C1-L506C14) :)